logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -135,13 +148,46 @@ public class AbstractPipelineExt extends RubyBasicObject {
     private QueueReadClientBase filterQueueClient;
 
     private ArrayList<FlowMetric> flowMetrics = new ArrayList<>();
+    private @SuppressWarnings("rawtypes") RubyArray inputs;
+    private @SuppressWarnings("rawtypes") RubyArray filters;
+    private @SuppressWarnings("rawtypes") RubyArray outputs;
 
     public AbstractPipelineExt(final Ruby runtime, final RubyClass metaClass) {
         super(runtime, metaClass);
     }
 
+    @JRubyMethod(required = 4)
+    public AbstractPipelineExt initialize(final ThreadContext context, final IRubyObject[] args)
```

Comment:
Can we have a documentation for this method? I really confuse whenever I do see at this. We have a caller in `reload.rb:L51`
```
LogStash::AbstractPipeline.new(@pipeline_config, nil, logger, nil)
```
If we do not use `args[1]` & `args[3]` as they are nil, why don't we remove?

Reply:
As stated in the PR, this is a net-zero-change PR, very intentionally designed to change only the hierarchy using automated code tools and as little manual intervention as possible.

I have very little idea what several of these pieces do, and we will have opportunity to refactor and consolidate separately.